File: examples/quickstart-firefox/src/main/java/org/fluentlenium/examples/pages/fluentlenium/MainPage.java
package org.fluentlenium.examples.pages.fluentlenium;

import static org.fluentlenium.assertj.FluentLeniumAssertions.assertThat;

import org.fluentlenium.core.FluentPage;
import org.fluentlenium.core.annotation.PageUrl;
import org.fluentlenium.core.domain.FluentWebElement;
import org.fluentlenium.examples.components.fluentlenium.Header;
import org.openqa.selenium.support.FindBy;

@PageUrl("https://fluentlenium.com")
public class MainPage extends FluentPage {

    @FindBy(className = "whats-fluentlenium")
    private FluentWebElement mainContent;

    @FindBy(css = "nav")
    private Header header;

    @Override
    public void isAt() {
        assertThat(mainContent).isDisplayed();
    }

    public Header getHeader() {
        return header;
    }

}

File: examples/quickstart-firefox/src/main/java/org/fluentlenium/examples/pages/fluentlenium/MainPage.kt
package org.fluentlenium.examples.pages.fluentlenium

import org.fluentlenium.assertj.FluentLeniumAssertions.assertThat
import org.fluentlenium.core.FluentPage
import org.fluentlenium.core.annotation.PageUrl
import org.fluentlenium.core.domain.FluentWebElement
import org.fluentlenium.examples.components.fluentlenium.Header
import org.openqa.selenium.support.FindBy

@PageUrl("https://fluentlenium.io")
class MainPage : FluentPage() {

    fun perform(fn: MainPage.() -> Unit) = this.apply(fn)

    @FindBy(className = "whats-fluentlenium")
    lateinit var mainContent: FluentWebElement

    @FindBy(css = "nav")
    lateinit var header: Header

    override fun isAt() {
        assertThat(mainContent).isDisplayed
    }

}